echo-of-the-song/Assets/Game/Scripts/Damage/Damageable.cs

```diff
@@ -0,0 +1,57 @@
+using Mechanics.Damage;
+using Mechanics.DetectorCollision;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Mechanics.HealtPoint
+{
+    public class Damageable : MonoBehaviour, IDamageable
+    {
+        [SerializeField]
+        private float health = 100;
+        [SerializeField]
+        private string tagEnemy = "Enemy";
+        [SerializeField]
+        private LayerMask layerMask;
+
+        public float Health
+        {
+            get => health;
+            set => health = value;
+        }
+
+        public void TakeDamage(float damage)
+        {
+            Health -= damage;
+
+            if (health < 0)
+            {
+                Die();
+            }
+        }
+
+        public void Die()
+        {
+            gameObject.SetActive(false);
+        }
+
+        public IEnumerable Detect()
```

Comment:
Why does IDamagable has a the detect method?
You should create separate class (SphereOverlapDetector : Monobehaviour, IDetector<IDamagable>) which will contain the Detect method : public IEnumerable<DetectionData<IDamagable>> Detect()